mountpoint-s3-client/src/object_client.rs

```diff
@@ -595,33 +618,20 @@ impl UploadChecksum {
 pub trait GetObjectResponse:
     Stream<Item = ObjectClientResult<GetBodyPart, GetObjectError, Self::ClientError>> + Send + Sync
 {
+    type BackpressureHandle: ClientBackpressureHandle;
     type ClientError: std::error::Error + Send + Sync + 'static;
 
+    /// Take the backpressure handle from the response.
+    ///
+    /// If `enable_read_backpressure` is false this call will return `None`,
+    /// no backpressure is being applied and data is being downloaded as fast as possible.
+    fn take_backpressure_handle(&mut self) -> Option<Self::BackpressureHandle>;
```

Comment:
Why `take_`? We should document that it can only be called once, or rather return a reference and make it `Clone`.